pygmt/tests/test_figure.py

```diff
@@ -377,13 +377,50 @@ class TestSetDisplay:
 
     def test_set_display(self):
         """
-        Test if pygmt.set_display updates the SHOW_CONFIG variable correctly.
+        Test if pygmt.set_display updates the SHOW_CONFIG variable correctly and
+        Figure.show opens the preview image in the correct way.
         """
-        default_method = SHOW_CONFIG["method"]  # Current default method
-
-        for method in ("notebook", "external", "none"):
-            set_display(method=method)
-            assert SHOW_CONFIG["method"] == method
+        default_method = SHOW_CONFIG["method"]  # Store the current default method.
+
+        fig = Figure()
+        fig.basemap(region=[0, 3, 6, 9], projection="X1c", frame=True)
+
+        # Test the "notebook" display method.
+        set_display(method="notebook")
+        assert SHOW_CONFIG["method"] == "notebook"
```

Comment:
Using `unittest.mock.patch`.

Need to note that we must pacth `pygmt.figure.launch_external_viewer` rather than `pygmt.helpers.utils.launch_external_viewer`.

See https://docs.python.org/3/library/unittest.mock.html#where-to-patch for the reasons:

> The basic principle is that you patch where an object is looked up, which is not necessarily the same place as where it is defined.